sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4367,6 +4367,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val JSON_USE_UNSAFE_ROW =
+    buildConf("spark.sql.json.useUnsafeRow")
+      .internal()
+      .doc("When set to true, use UnsafeRow to represent struct result in the JSON parser.")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
Wonder if we should enable this by default. If this benefits us in most of the cases, should be fine to enable it by default

Reply:
I feel it is better to not enable it by default. The benchmark has shown that applying the change can slow down the JSON scan a bit. If there is enough memory, then saving memory doesn't have enough benefit.